woocommerce/woocommerce
# Reduce duplicated code in Analytics classes

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This is a feature branch to gather multiple improvements to the Analytics codebase with the goal to:
- Make it easier for extension developers to reuse & extend
- Make it easier for our Team to maintain


- https://github.com/woocommerce/woocommerce/pull/48964
- https://github.com/woocommerce/woocommerce/pull/49320
- https://github.com/woocommerce/woocommerce/pull/49427
- https://github.com/woocommerce/woocommerce/pull/50746
- https://github.com/woocommerce/woocommerce/pull/50748
- https://github.com/woocommerce/woocommerce/pull/50749
- https://github.com/woocommerce/woocommerce/pull/50799



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

All PRs were reviewed separately, but to double-check

1. Smoke test analytics for regression
2. Review the code once again

Apart from PRs, I pushed one commit to set the target release version in `since`tags https://github.com/woocommerce/woocommerce/pull/49425/commits/e3286534e6e96135858e2b0712c1dfce6851279c



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
